src/status_im2/contexts/quo_preview/share/qr_code.cljs

```diff
@@ -2,22 +2,56 @@
   (:require [quo2.foundations.colors :as colors]
             [react-native.core :as rn]
             [quo2.core :as quo]
-            [status-im2.common.resources :as resources]))
+            [utils.image-server :as image-server]
+            [utils.re-frame :as rf]
+            [reagent.core :as reagent]
+            [status-im2.contexts.quo-preview.preview :as preview]))
+
+(def descriptor
+  [{:label "URL For QR"
+    :key   :text
+    :type  :text}
+   {:label   "Error Correction Level:"
+    :key     :error-correction-level
+    :type    :select
+    :options [{:key   :low
+               :value "Low"}
+              {:key   :medium
+               :value "Medium"}
+              {:key   :quart
+               :value "Quart"}
+              {:key   :highest
+               :value "Highest"}]}
+  ])
 
 (defn cool-preview
   []
-  (fn []
-    [rn/touchable-without-feedback {:on-press rn/dismiss-keyboard!}
-     [rn/view {:style {:padding-bottom 150}}
-      [rn/view {:style {:flex 1}}]
-      [rn/view
-       {:style {:padding-vertical 60
-                :flex-direction   :row
-                :justify-content  :center}}
-       [quo/qr-code
-        {:source (resources/get-mock-image :qr-code)
-         :height 250
-         :width  250}]]]]))
+  (let [state                  (reagent/atom {:text                   "https://status.im"
+                                              :error-correction-level :highest})
+        text                   (reagent/cursor state [:text])
+        error-correction-level (reagent/cursor state [:error-correction-level])
```

Comment:
I like the usage of cursors in these preview screens, I'll use your idea here in the future.